root: new version 6.28.00

wdconinc opened the pull request:
Full changelog: https://github.com/root-project/root/compare/v6-26-10...v6-28-00

Based on the CMakeLists.txt file, no major build system changes. Upgraded requirements for cmake added.

wdconinc committed:
root: new version 6.28.00

Full changelog: https://github.com/root-project/root/compare/v6-26-10...v6-28-00

Based on the CMakeLists.txt file, no major build system changes. Upgraded requirements for cmake added.

HadrienG2 reviewed (changes requested):
I see in the release notes that "The build options gfal, gsl_shared, jemalloc, monalisa, pyroot_legacy, tcmalloc, and xproofd have been deprecated. Please complain with root-dev@cern.ch should you still need one!".

On our side, we expose the jemalloc variant, which we'll probably want to mark as deprecated accordingly, and we internally use gsl_shared when the gsl variant is enabled, so we need to investigate why we use it and how bad losing it will be.

wdconinc commented:
There is no way to deprecate variants in spack, AFAIK. I propose something like:
```diff
-    variant("jemalloc", default=False, description="Enable using the jemalloc allocator")
+    variant("jemalloc", when"@:6.28", default=False, description="Enable using the jemalloc allocator (deprecated in 6.28)")
```

The `gsl_shared` build option doesn't seem to do anything at all from what I can tell... I can't find any useful references until the git history of the root5 days.

HadrienG2 commented:
I like your proposed solution for the jemalloc variant. Concerning `gsl_shared`, if it doesn't do anything, I would say let's try to disable it (at least if `spec.satisfies("@6.28:")`, but if it's been useless for so long, maybe perma-set it to False for all releases?) and see if users report back any problem.

wdconinc committed:
root: variant jemalloc deprecated

wdconinc commented:
For the oldest version, all I can find is that `gsl_shared` may simply have been inherited from `./configure` but not implemented in CMake.
```console
12:46:00 wdconinc@menelaos ~/git/root ((tags/v6-06-02^0) %) $ grep -iIr gsl_shared
cmake/modules/RootBuildOptions.cmake:ROOT_BUILD_OPTION(gsl_shared OFF "Enable linking against shared libraries for GSL (default no)")
configure:   enable_gsl_shared         \
configure:enable_gsl_shared=no
configure:    # enable_gsl_shared=$enable_shared"
configure:        check_library "libgsl gslML gsl" "$enable_gsl_shared" "$gsllibdir" \
configure:                    "$enable_gsl_shared" "" \
configure:                if test "x$enable_gsl_shared" = "xyes"; then
configure:       test "x$enable_gsl_shared" = "xno" ; then
```

wdconinc commented:
gsl_shared introduced here: https://github.com/root-project/root/commit/f75abd6e6e801baddfa45bc906763c62e005c02f

and "gsl_shared is unused" per https://github.com/root-project/root/pull/12203

wdconinc committed:
root: rm define gsl_shared because unused

"`gsl_shared` is unused" per https://github.com/root-project/root/pull/12203 (and has always been unused)

wdconinc commented:
@spackbot fix style

wdconinc committed:
[@spackbot] updating style on behalf of wdconinc

tldahlgren commented on var/spack/repos/builtin/packages/root/package.py:
```diff
@@ -505,7 +513,6 @@ def cmake_args(self):
             define_from_variant("gl2ps", "opengl"),
             define("glite", False),
             define("globus", False),
-            define_from_variant("gsl_shared", "gsl"),
```
Just to confirm: removing this option for *all* version installs was deliberate, correct?

wdconinc replied:
It never did anything.

wdconinc replied:
Not in this millenium anyway.

tldahlgren reviewed (approved):
Confirmed the sha256.  Just a quick question about removal of one of the options.
